libs/database/src/lib/services/notification.service.v2.ts

```diff
@@ -0,0 +1,457 @@
+
+    if (notify.type === NotificationType.System)
+      message = CastcleLocalization.getTemplateSystem(language, displayNames);
+
+    if (notify.type === NotificationType.AdsApprove)
+      message = CastcleLocalization.getTemplateAdsApprove(language);
+
+    if (notify.type === NotificationType.AdsDecline)
+      message = CastcleLocalization.getTemplateAdsDecline(language);
+
+    if (notify.type === NotificationType.Follow)
+      message = CastcleLocalization.getTemplateFollow(language, displayNames);
+
+    this.#logger.log('Prepare message show display name.', message);
+    return message;
+  };
+
+  generateMessagesToNotifications = async (
+    notifies: Notification[],
+    language: string
+  ) => {
+    this.#logger.log('Start generate notification message list.');
+
+    return await Promise.all(
+      notifies.map(async (notify: Notification) => {
+        const usersOwners = await this._userModel
+          .find({
+            ownerAccount: notify.account,
+          })
+          .exec();
+
+        const userSource = usersOwners.filter(
+          (item) =>
+            item.type === UserType.PAGE ||
+            (item.type === 'people' &&
+              notify.source === NotificationSource.Profile)
+        );
+
+        const userIds = notify.sourceUserId?.reverse();
+        const users = await this._userModel
+          .find({
+            _id: { $in: userIds },
+          })
+          .exec();
+        const userSort = [];
+        users.forEach((user) => {
+          const index = userIds.indexOf(user._id);
+          if (index > -1) userSort[index] = user;
+        });
+
+        let message = '';
+        const displayNames = userSort.map((user) => user.displayName);
+
+        if (notify.type === NotificationType.Like)
+          if (notify.commentRef) {
+            message = CastcleLocalization.getTemplateLikeComment(
+              language,
+              displayNames,
+              userSource[0]?.type === UserType.PAGE
+                ? userSource[0]?.displayName
+                : ''
+            );
+          } else {
+            message = CastcleLocalization.getTemplateLike(
+              language,
+              displayNames,
+              userSource[0]?.type === UserType.PAGE
+                ? userSource[0]?.displayName
+                : ''
+            );
+          }
+        if (notify.type === NotificationType.Comment)
+          message = CastcleLocalization.getTemplateComment(
+            language,
+            displayNames,
+            userSource[0]?.type === UserType.PAGE
+              ? userSource[0]?.displayName
+              : ''
+          );
+        if (notify.type === NotificationType.Farm)
+          message = CastcleLocalization.getTemplateFarm(
+            language,
+            displayNames,
+            userSource[0]?.type === UserType.PAGE
+              ? userSource[0]?.displayName
+              : ''
+          );
+        if (notify.type === NotificationType.Quote)
```

Comment:
Can you refactor to use switch case instead.

Switch(notify.type){
 case NotificationType.Quote:
 case NotificationType.Farm:
     message = CastcleLocalization.getTemplateFarm(
             language,
             displayNames,
             userSource[0]?.type === UserType.PAGE
               ? userSource[0]?.displayName
               : ''
           );
           break;
      }